src/hotspot/share/gc/g1/g1CardSetContainers.hpp

```diff
@@ -124,6 +124,11 @@ class G1CardSetContainer {
 public:
   G1CardSetContainer() : _ref_count(3) { }
 
+  static G1CardSetContainer* volatile* next_ptr(G1CardSetContainer& node) {
+    return node.next_addr();
+  }
+  typedef LockFreeStack<G1CardSetContainer, &next_ptr> NodeStack;
+
```

Comment:
I really dislike making these implementation details public.  At the very least, I'd like a comment or conditionalization or something to warn folks away from using them, with an expiration cleanup RFE or something.  An unannotated compiler workaround runs the risk of someone coming along and cleaning it up, cycling through this again.  Or confusing future readers when the need for the workaround is long over.

Reply:
Not relevant anymore.